contributingGuides/STYLE.md

```diff
@@ -10,13 +68,497 @@ We use Prettier to automatically style our code.
+    getHello: () => string;
+    getGoodbye: () => string;
+};
+
+// index.native.ts
+import { GreetingModule } from "./types";
+function getHello() {
+    return "hello from mobile code";
+}
+function getGoodbye() {
+    return "goodbye from mobile code";
+}
+const Greeting: GreetingModule = {
+    getHello,
+    getGoodbye,
+};
+export default Greeting;
+
+// index.ts
+import { GreetingModule } from "./types";
+function getHello() {
+    return "hello from other platform code";
+}
+function getGoodbye() {
+    return "goodbye from other platform code";
+}
+const Greeting: GreetingModule = {
+    getHello,
+    getGoodbye,
+};
+export default Greeting;
+```
+
+Component module example
+
+```ts
+// types.ts
+export type MyComponentProps = {
+    foo: string;
+}
+
+// index.ios.ts
+import { MyComponentProps } from "./types";
+
+export MyComponentProps;
+export default function MyComponent({ foo }: MyComponentProps) { /* ios specific implementation */ }
+
+// index.ts
+import { MyComponentProps } from "./types";
+
+export MyComponentProps;
+export default function MyComponent({ foo }: MyComponentProps) { /* Default implementation */ }
+```
+
+### Reusable Types
+
+Reusable type definitions, such as models (e.g., Report), must have their own file and be placed under `src/types/`. The type should be exported as a default export.
+
+```ts
+// src/types/Report.ts
+
+type Report = {...};
+
+export default Report;
+```
+
+### `tsx` extension
+
+Use `.tsx` extension for files that contain React syntax.
+
+> Why? It is a widely adopted convention to mark any files that contain React specific syntax with `.jsx` or `.tsx`.
+
+### No inline prop types
+
+Do not define prop types inline for components that are exported.
+
+> Why? Prop types might [need to be exported from component files](#export-prop-types). If the component is only used inside a file or module and not exported, then inline prop types can be used.
+
+```ts
+// BAD
+export default function MyComponent({ foo, bar }: { foo: string, bar: number }){
+   // component implementation
+};
+
+// GOOD
+type MyComponentProps = { foo: string, bar: number };
+export default MyComponent({ foo, bar }: MyComponentProps){
+   // component implementation
+}
+```
+
+### Satisfies Operator
+
+Use the `satisfies` operator when you need to validate that the structure of an expression matches a specific type, without affecting the resulting type of the expression.
+
+> Why? TypeScript developers often want to ensure that an expression aligns with a certain type, but they also want to retain the most specific type possible for inference purposes. The `satisfies` operator assists in doing both.
+
+```ts
+// BAD
+const sizingStyles = {
+    w50: {
+        width: '50%',
+    },
+    mw100: {
+        maxWidth: '100%',
+    },
+} as const;
+
+// GOOD
+const sizingStyles = {
+    w50: {
+        width: '50%',
+    },
+    mw100: {
+        maxWidth: '100%',
+    },
+} satisfies Record<string, ViewStyle>;
```

Comment:
*question:* Would you need to do this to get the most narrow type?

```suggestion
} as const satisfies Record<string, ViewStyle>;
```

Reply:
I think so, but @fabioh8010 @blazejkustra can you confirm?

Reply:
> _question:_ Would you need to do this to get the most narrow type?

Yes that is correct, we could put another example as there are cases `as const satisfies` is not desirable.

Reply:
I think it makes a difference in some cases, example:

satisfies Record<string, ViewStyle>:
```ts
    br0: {
        borderRadius: number;
    };
    br1: {
        borderRadius: number;
    };
}
```
as const satisfies Record<string, ViewStyle>:
```ts
{
    readonly br0: {
        readonly borderRadius: 0;
    };
    readonly br1: {
        readonly borderRadius: 4;
    };
}
```

So if we see any value in such narrow type I'm good to adjust the guidelines 👍